src/orion/algo/evolution_es.py

```diff
@@ -222,12 +228,12 @@ def get_candidates(self, rung_id):
                     point[j] = self.eves.population[j][i]
 
                 if tuple(point) in points:
-                    nums_all_equal += 1
+                    nums_all_equal[i] += 1
                     logger.debug("find equal one, continue to mutate.")
                     self._mutate(points.index(tuple(point)), i)
```

Comment:
I would be in favor of moving these mutations to avoid duplicates to `_mutate_population` so that roles of `_mutate_population` and `_get_mutated_points` are clearly separated. If I understand correctly this function would be only returning the population in the format of points. So I would suggest to rename it `_get_population_points`. (Assuming the mutation to avoid duplications are moved to `_mutate_population`). This would also avoid side-effects in this function, concentrating all in-place operations in `_mutate_population`.

Reply:
In my opinion, there is no need to add a function to merely implement "convert population to points" (there may be only one sentence to do that.) So I move the duplications into _mutate_population() as you suggestion, and implement the "convert population to points" in passing.

Reply:
I would be in favor of keeping them separated to simplify testing, but let's keep them together now that tests are ok.